File: 0242-valid-anagram/0242-valid-anagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        
        if len(s) != len(t):
            return False
        
        dict_s = dict()
        dict_t = dict()
        
        for value in s:
            if value in dict_s:
                dict_s[value] = dict_s[value] + 1
            else:
                dict_s[value] = 1;
            
        for value in t:
            if value in dict_t:
                dict_t[value] = dict_t[value] + 1
            else:
                dict_t[value] = 1;
        
        for key in dict_s:
            if key not in dict_t:
                return False
            elif dict_t[key] != dict_s[key]:
                return False
            
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        
        if(s.length() != t.length()){
            return false;
        }
        
        char[] s_arr = s.toCharArray();
        char[] t_arr = t.toCharArray();
        
        HashMap<Character, Integer> map = new HashMap<>();
        
        for(char c : s_arr){
            int c_cnt = map.get(c) == null ? 0 : map.get(c);
            map.put(c, c_cnt + 1);
        }
        
        for(char c : t_arr){
            int c_cnt = map.get(c) == null ? 0 : map.get(c);
            if(c_cnt == 0){
                return false;
            }else{
                map.put(c, c_cnt - 1);
            }
        }
        
        return true;
    }
}